omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_8/ConceptResource1_8.java

```diff
@@ -516,16 +517,30 @@ public void purge(Concept concept, RequestContext context) throws ResponseExcept
 		Context.getConceptService().purgeConcept(concept);
 	}
 	
-	/**
-	 * This does not include retired concepts
-	 * 
-	 * @see org.openmrs.module.webservices.rest.web.resource.impl.DelegatingCrudResource#doGetAll(org.openmrs.module.webservices.rest.web.RequestContext)
-	 */
 	@Override
-	protected NeedsPaging<Concept> doGetAll(RequestContext context) {
-		List<Concept> allConcepts = Context.getConceptService().getAllConcepts(null, true, context.getIncludeAll());
-		return new NeedsPaging<Concept>(allConcepts, context);
-	}
+	protected AlreadyPaged<Concept> doGetAll(RequestContext context){
+		List<Locale> locales = new ArrayList<Locale>();
+		locales.add(Context.getLocale());
+		
+		List<Concept> result = Context.getConceptService().getAllConcepts();
+		List<Concept> concepts = new ArrayList<Concept>();
+		   for (Concept concept : result) {
+			    concepts.addAll((Collection<? extends Concept>) concept.getConceptClass());
+			    StringBuilder sb = new StringBuilder("result");
+			    for(Concept concept1:result) {
```

Comment:
Hey whats the reason to iterate on the same list 'result' twice ? Also inside the second for loop you are creating a string which is not used any where ?